src/api/routes/order_routes.py

```
@@ -0,0 +1,114 @@
+    
+    @marshal_with(order_schema)
+    def get(self, order_id):
+        """Get an order by id."""
+        order = Order.query.get_or_404(order_id)
+        return order, 200
+    
+    @marshal_with(order_schema)
+    def post(self):
+        """Create a new order and queue it for processing."""
+        parser = reqparse.RequestParser()
+        parser.add_argument(
+            'hero_id', 
+            type=int, 
+            required=True, 
+            help='Hero_id is required',
+            location=['form', 'json']
+        )
+        parser.add_argument(
+            'meal_id', 
+            type=int, 
+            required=True, 
+            help='Meal_D is required',
+            location=['form', 'json']
+        )
+        args = parser.parse_args()
+
+        # hero and meal exist?
+        hero = Hero.query.get(args['hero_id'])
+        if not hero:
+            return {'message': f"Hero with ID {args['hero_id']} not found"}, 404
+        
+        meal = Meal.query.get(args['meal_id'])
+        if not meal:
+            return {'message': f"Meal with ID {args['meal_id']} not found"}, 404
+        
+        # Create order with PENDING status
+        order = Order(
+            hero_id=args['hero_id'],
+            meal_id=args['meal_id'],
+            status=OrderStatus.PENDING,
+            order_time=datetime.utcnow()
+        )
+        
+        db.session.add(order)
+        db.session.commit()
+        
+        # Send task to Redis queue
+        try:
+            redis_url = current_app.config.get('REDIS_URL', 'redis://tcc-redis:6379/0')
+            redis_conn = redis.from_url(redis_url)
+            
+            # Create queue
+            queue = Queue('order_processing', connection=redis_conn)
+            
+            # Enqueue the task
+            queue.enqueue('src.worker.tasks.process_order', order.id, job_timeout=300)
+            
+            return order, 202
+        
+        except Exception as e:
```

The exception handling is too broad. Catching all exceptions with a generic Exception can hide unexpected errors and make debugging more difficult.

It is misleading to return a 202 status code when the task was not actually enqueued. This gives the impression that everything went well.

You should catch only the specific exceptions you expect (such as Redis-related errors), and handle them accordingly.

If enqueueing the task fails, a more appropriate response code would be 503 (Service Unavailable), not 202.

Logging is good, but additional context in the logs (e.g. Redis URL, job parameters) could be helpful for diagnosing failures.

Consider separating concerns: the Redis connection, queue creation, and task enqueueing could each have their own try/except block for better clarity and error handling.

If enqueueing fails, you might also consider implementing a retry mechanism or flagging the order for later reprocessing.